src/components/Kanban/Kanban.tsx

```diff
@@ -1,34 +1,119 @@
-import React, { MouseEventHandler, useCallback } from 'react';
+import React, { FC, MouseEventHandler, useCallback, useEffect, useMemo, useRef, useState } from 'react';
 import { KanbanColumn, KanbanContainer } from '@taskany/bricks/harmony';
+import { nullable, useIntersectionLoader } from '@taskany/bricks';
 
 import { trpc } from '../../utils/trpcClient';
-import { FilterById } from '../../../trpc/inferredTypes';
+import { FilterById, State } from '../../../trpc/inferredTypes';
+import { refreshInterval } from '../../utils/config';
 import { useUrlFilterParams } from '../../hooks/useUrlFilterParams';
+import { useGoalPreviewInvalidate } from '../../hooks/useGoalPreviewInvalidate';
 import { NextLink } from '../NextLink';
 import { State as KanbanState } from '../State';
 import { GoalsKanbanCard } from '../GoalsKanbanCard/GoalsKanbanCard';
 import { useGoalPreview } from '../GoalPreview/GoalPreviewProvider';
 import { routes } from '../../hooks/router';
+import { Loader } from '../Loader/Loader';
+import { NoGoalsText } from '../NoGoalsText/NoGoalsText';
 
 import s from './Kanban.module.css';
 
-type KanbanItemProps = Omit<
-    React.ComponentProps<typeof GoalsKanbanCard>,
-    keyof React.HTMLAttributes<HTMLDivElement>
-> & { id: string; shortId: string; title: string };
+type LoadingState = 'loading' | 'ready' | 'loaded' | 'empty';
+type onLoadingStateChange = (state: LoadingState) => void;
 
-interface KanbanProps {
-    value: Record<string, KanbanItemProps[]>;
+interface KanbanStateColumnProps {
+    state: State;
+    projectId: string;
     filterPreset?: FilterById;
+    partnershipProject?: string[];
+    onLoadingStateChange?: onLoadingStateChange;
 }
 
-export const Kanban = ({ value, filterPreset }: KanbanProps) => {
-    const { data: states = [] } = trpc.state.all.useQuery();
+const calculateCommonLoadingState = (map: Map<unknown, LoadingState>): LoadingState => {
```

Comment:
Why is it so difficult? Could you're move state calculations logic on column level? Cause that level has fetching the data